src/core/deltafilewrapper.cpp

```diff
@@ -892,17 +778,213 @@ void DeltaFileWrapper::addCreate( const QString &localLayerId, const QString &so
 
   delta.insert( QStringLiteral( "new" ), newData );
 
-  QString localPk = delta.value( QStringLiteral( "localPk" ) ).toString();
+  appendDelta( delta );
+}
+
+void DeltaFileWrapper::appendDelta( const QJsonObject &delta )
+{
+  if ( mIsPushing )
+  {
+    mPendingDeltas << delta;
+  }
+  else
+  {
+    mergeDelta( delta );
+  }
+}
+
+void DeltaFileWrapper::mergeCreateDelta( const QJsonObject &delta )
+{
+  const QString localPk = delta.value( QStringLiteral( "localPk" ) ).toString();
```

Comment:
```suggestion
  Q_ASSERT( delta.value( QStringLiteral( "method" ) ) == "create" );

  const QString localPk = delta.value( QStringLiteral( "localPk" ) ).toString();
```

Just being paranoid.

Reply:
I don't think we ever build "Debug" builds where this would trigger.
We could consider to define our own `QF_ASSERT` that reports to sentry